packages/openid4vc/src/shared/utils.ts

```diff
@@ -34,61 +38,95 @@ export function getSupportedJwaSignatureAlgorithms(agentContext: AgentContext):
   return supportedJwaSignatureAlgorithms
 }
 
-export async function getSphereonSuppliedSignatureFromJwtIssuer(
-  agentContext: AgentContext,
-  jwtIssuer: OpenId4VcJwtIssuer
-): Promise<SuppliedSignature> {
-  let key: Key
-  let alg: string
-  let kid: string | undefined
-  let did: string | undefined
-
-  if (jwtIssuer.method === 'did') {
-    const didsApi = agentContext.dependencyManager.resolve(DidsApi)
-    const didDocument = await didsApi.resolveDidDocument(jwtIssuer.didUrl)
-    const verificationMethod = didDocument.dereferenceKey(jwtIssuer.didUrl, ['authentication'])
-
-    // get the key from the verification method and use the first supported signature algorithm
-    key = getKeyFromVerificationMethod(verificationMethod)
-    const _alg = getJwkClassFromKeyType(key.keyType)?.supportedSignatureAlgorithms[0]
-    if (!_alg) throw new CredoError(`No supported signature algorithms for key type: ${key.keyType}`)
+async function getKeyFromDid(agentContext: AgentContext, didUrl: string) {
+  const didsApi = agentContext.dependencyManager.resolve(DidsApi)
+  const didDocument = await didsApi.resolveDidDocument(didUrl)
+  const verificationMethod = didDocument.dereferenceKey(didUrl, ['authentication'])
+
+  return getKeyFromVerificationMethod(verificationMethod)
+}
+
+export function getVerifyJwtCallback(agentContext: AgentContext): VerifyJwtCallback {
+  return async (jwtVerifier, jwt) => {
+    const jwsService = agentContext.dependencyManager.resolve(JwsService)
+    if (jwtVerifier.method === 'did') {
+      const key = await getKeyFromDid(agentContext, jwtVerifier.didUrl)
+      const jwk = getJwkFromKey(key)
 
-    alg = _alg
-    kid = verificationMethod.id
-    did = verificationMethod.controller
-  } else {
-    throw new CredoError(`Unsupported jwt issuer method '${jwtIssuer.method as string}'. Only 'did' is supported.`)
+      const res = await jwsService.verifyJws(agentContext, { jws: jwt.raw, jwkResolver: () => jwk })
+      return res.isValid
+    } else if (jwtVerifier.method === 'x5c' || jwtVerifier.method === 'jwk') {
+      const res = await jwsService.verifyJws(agentContext, { jws: jwt.raw })
+      return res.isValid
+    } else {
+      throw new Error(`Unsupported jwt verifier method: '${jwtVerifier.method}'`)
+    }
   }
+}
+
+export function getCreateJwtCallback(agentContext: AgentContext): CreateJwtCallback {
+  return async (jwtIssuer, jwt) => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { iss, sub, aud, exp, nbf, iat, jti, ...additionalClaims } = jwt.payload
+
+    const jwsService = agentContext.dependencyManager.resolve(JwsService)
+
+    if (jwtIssuer.method === 'did') {
+      const key = await getKeyFromDid(agentContext, jwtIssuer.didUrl)
+      const jws = await jwsService.createJwsCompact(agentContext, {
+        protectedHeaderOptions: { alg: jwtIssuer.alg, ...jwt.header },
+        payload: new JwtPayload({ ...jwt.payload, additionalClaims }),
```

Comment:
You can use JwtPayload.fromJson here? so you don't have to spread all the properties